### PR TITLE
Use eventBus to select rows

### PR DESCRIPTION
**Pull request by amanmahajan7**
As`handleRowSelectionChange` is dependent on `selectedRows`, the whole grid is re-rendered when a row is selected.

**Before**
image

**After**

image

### CODE REVIEW

File: src/EventBus.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
 import { Position } from './common/types';
 
+export interface SelectRowEvent<R> {
```
Comment (nstepien) [inline src/EventBus.ts]: Should probably move this in our `types` file.

File: src/Cell.tsx [hunk 2]
```diff
@@ -141,7 +146,7 @@ function Cell<R>({
       rowIdx,
       row: rowData,
       isRowSelected,
-      onRowSelectionChange,
+      onRowSelectionChange: isSummaryRow ? noop : onRowSelectionChange,
```
Comment (nstepien) [inline src/Cell.tsx]: fwiw our checkbox column doesn't display any checkbox for summary rows, so maybe we don't need this check here.
And maybe users might want to display set up a custom column/row selection behavior.

File: src/Canvas.tsx [hunk 3]
```diff
@@ -155,6 +158,34 @@ function Canvas<R, K extends keyof R>({
     selectCell({ rowIdx, idx }, true);
   }
 
+  useEffect(() => {
```
Comment (nstepien) [inline src/Canvas.tsx]: Any particular reason to move this from DataGrid to Canvas?

Comment (amanmahajan7) [reply src/Canvas.tsx]: Need access to `eventBus` which is initialized in the Canvas component

File: src/Cell.tsx (hunk 2 above)
Comment (amanmahajan7) [reply src/Cell.tsx]: Makes sense, I have removed it

**Pull request merged**